util/env_test.cc

```diff
@@ -923,41 +919,38 @@ TEST_P(EnvPosixTestWithParam, InvalidateCache) {
     // Random Read
     {
       unique_ptr<RandomAccessFile> file;
-      char scratch[kSectorSize];
+      auto scratch = NewAligned(kSectorSize, 0);
       Slice result;
 #if !defined(OS_MACOSX) && !defined(OS_WIN)
       if (soptions.use_direct_reads) {
-        rocksdb::SyncPoint::GetInstance()->SetCallBack(
-            "NewRandomAccessFile:O_DIRECT", [&](void* arg) {
-              int* val = static_cast<int*>(arg);
-              *val &= ~O_DIRECT;
-            });
+        soptions.use_direct_reads = false;
```

Comment:
i read that open() with O_DIRECT on tmpfs will return EINVAL. can you just check for that?